[SPARK-37272][TESTS] Add `ExtendedRocksDBTest` and disable RocksDB tests on Apple Silicon

dongjoon-hyun opened the pull request:
### What changes were proposed in this pull request?

This PR aims to add `ExtendedRocksDBTest` to disable `RocksDB` selectively on Apple Silicon.
- Add `ExtendedRocksDBTest` test tag
- Disable RocksDB test cases from `StreamingSessionWindowSuite`.

### Why are the changes needed?

Since `RocksDBJNI` still doesn't support Apple Silicon natively, the following failures occur on M1.

**BEFORE**
```
$ build/sbt "sql/testOnly *RocksDB* *.StreamingSessionWindowSuite"
...
[info] Run completed in 23 seconds, 281 milliseconds.
[info] Total number of tests run: 32
[info] Suites: completed 2, aborted 2
[info] Tests: succeeded 22, failed 10, canceled 0, ignored 0, pending 0
[info] *** 2 SUITES ABORTED ***
[info] *** 10 TESTS FAILED ***
[error] Failed tests:
[error] 	org.apache.spark.sql.streaming.StreamingSessionWindowSuite
[error] 	org.apache.spark.sql.execution.streaming.state.RocksDBStateStoreIntegrationSuite
[error] Error during tests:
[error] 	org.apache.spark.sql.execution.streaming.state.RocksDBSuite
[error] 	org.apache.spark.sql.execution.streaming.state.RocksDBStateStoreSuite
[error] (sql / Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 43 s, completed Nov 10, 2021 4:29:50 PM
```

**AFTER**
```
$ build/sbt "sql/testOnly *RocksDB* *.StreamingSessionWindowSuite" -Dtest.exclude.tags=org.apache.spark.tags.ExtendedRocksDBTest
...
[info] Run completed in 21 seconds, 353 milliseconds.
[info] Total number of tests run: 14
[info] Suites: completed 4, aborted 0
[info] Tests: succeeded 14, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 41 s, completed Nov 10, 2021 4:37:01 PM
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually test on Apple Silicon (M1/M1 Pro/M1 Max).

dongjoon-hyun committed:
[SPARK-37272][TESTS] Add ExtendedRocksDBTest

dongjoon-hyun commented:
Could you review this, @viirya ?

SparkQA commented:
**[Test build #145073 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/145073/testReport)** for PR 34547 at commit [`f422fea`](https://github.com/apache/spark/commit/f422fea8f6fd1abb765c433cda64b68d5a64d4e5).

HyukjinKwon commented on sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingSessionWindowSuite.scala:
```diff
@@ -43,12 +43,17 @@ class StreamingSessionWindowSuite extends StreamTest
+    // RocksDB doesn't support Apple Silicon yet
+    if (System.getProperty("os.name").equals("Mac OS X") &&
+        System.getProperty("os.arch").equals("aarch64")) {
+      providerOptions = providerOptions.take(1)
```
nit but maybe it would be better to explicitly filter `RocksDBStateStoreProvider` out

dongjoon-hyun replied:
Sure.

HyukjinKwon commented on sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingSessionWindowSuite.scala:
```diff
@@ -43,12 +43,17 @@ class StreamingSessionWindowSuite extends StreamTest
     ).map { value =>
       (SQLConf.STATE_STORE_PROVIDER_CLASS.key, value.stripSuffix("$"))
     }
+    // RocksDB doesn't support Apple Silicon yet
```
Maybe we should document this somewhere but could be done separately ..

dongjoon-hyun replied:
Yes, definitely. I'm currently working on this area and will do the similar with LevelDB.
I'll document it both together.

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/49542/

dongjoon-hyun committed:
address comments

dongjoon-hyun commented:
I addressed the comment by 
```scala
- providerOptions = providerOptions.take(1)
+ providerOptions = providerOptions
+   .filterNot(_._2.contains(classOf[RocksDBStateStoreProvider].getSimpleName))
```

And, tested like the following.
```
$ build/sbt "sql/testOnly *RocksDB* *.StreamingSessionWindowSuite" -Dtest.exclude.tags=org.apache.spark.tags.ExtendedRocksDBTest
...
[info] Run completed in 18 seconds, 360 milliseconds.
[info] Total number of tests run: 14
[info] Suites: completed 4, aborted 0
[info] Tests: succeeded 14, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 30 s, completed Nov 10, 2021 6:37:02 PM
```

viirya commented:
lgtm

SparkQA commented:
Kubernetes integration test status failure
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/49542/

dongjoon-hyun commented:
Thank you, @HyukjinKwon and @viirya !

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/49542/

SparkQA commented:
**[Test build #145073 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/145073/testReport)** for PR 34547 at commit [`f422fea`](https://github.com/apache/spark/commit/f422fea8f6fd1abb765c433cda64b68d5a64d4e5).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/145073/
